core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -148,10 +148,19 @@ class AdminClient(val time: Time,
       throw new IllegalArgumentException(s"Consumer Group $groupId with protocol type '${metadata.protocolType}' is not a valid consumer group")
 
     Errors.forCode(metadata.errorCode()).maybeThrow()
-    val consumers = metadata.members.map { consumer =>
-      val assignment = ConsumerProtocol.deserializeAssignment(ByteBuffer.wrap(Utils.readBytes(consumer.memberAssignment)))
-      ConsumerSummary(consumer.memberId, consumer.clientId, consumer.clientHost, assignment.partitions.toList)
-    }.toList
+    val consumers = metadata.state match {
+      case "Stable" =>
+        metadata.members.map { consumer =>
+          val assignment = ConsumerProtocol.deserializeAssignment(ByteBuffer.wrap(Utils.readBytes(consumer.memberAssignment)))
+          ConsumerSummary(consumer.memberId, consumer.clientId, consumer.clientHost, assignment.partitions.toList)
+        }.toList
+      case "PreparingRebalance" | "AwaitingSync" =>
```

Comment:
> A little bikeshedding.. It seems less than ideal to mix consumers with no assignment in the same output. An alternative would be to split the current --describe into two separate options, say --members and --assignment.

I'm fine with doing this, but I'd suggest doing it in a separate JIRA (and hash out further details) so we can merge this PR as it's breaking the tests.